src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs

```diff
@@ -166,20 +165,20 @@ ValueTask<long> SerializeTimeoutException(IJsonRpcService service, Stream result
                     try
                     {
                         JsonRpcContext jsonRpcContext = JsonRpcContext.Http(jsonRpcUrl);
+                        long totalResponseSize = 0;
                         await foreach (JsonRpcResult result in jsonRpcProcessor.ProcessAsync(request, jsonRpcContext))
                         {
-                            Stream resultStream = jsonRpcConfig.BufferResponses ? new MemoryStream() : ctx.Response.Body;
+                            Stream stream = jsonRpcConfig.BufferResponses ? new MemoryStream() : null;
+                            ICountingBufferWriter resultWriter = stream is not null ? new CountingStreamPipeWriter(stream) : new CountingPipeWriter(ctx.Response.BodyWriter);
```

Comment:
You can potentially merge this into one, but you access stream later

Reply:
Need the stream to write later when buffered